src/lua/api-gateway/util/logger.lua

```diff
@@ -67,6 +67,23 @@ local function _decorateLogger()
     end
 end
 
+logger = {}
+
+function logger.error(...)
+    _decorateLogger()
```

Comment:
no need to decorage logger every time it's done once on _validateRequest to ensure that it is decorated across the request.